### PR TITLE
Add ES config for requests API

### PR DESCRIPTION
**Pull request by kenoir**
Currently the requests API has incorrect ES configuration, this changes adds that config.

### CODE REVIEW

File: terraform/shared/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+At present you will need to run a `terraform apply` operation, and then run the scripts in the `./scripts` folder.
+
+Run the scripts in this order:
+
+- [./scripts/create_elastic_users_catalogue.py](./scripts/create_elastic_users_catalogue.py): Creates roles and users for catalogue-api services, along with any other cluster configuration that doesn't make sense to apply in Terraform.
+- [./scripts/create_elastic_users_identity.py](./scripts/create_elastic_users_identity.py): Creates the users required in the identity account (specifically at present only the requests API).
+
+## Improvements
+
+We have used a local-exec provisioner in terraform to run these scripts, but adding users safely without rotating existing passwords is tricky. We should find a way to automate this safely from terraform.
```
Comment (alexwlchan) [inline terraform/shared/README.md:16]:
```suggestion
We have used a local-exec provisioner in terraform to run these scripts in the past, but adding users safely without rotating existing passwords is tricky. We should find a way to automate this safely from terraform.
```

File: terraform/shared/scripts/create_elastic_users_catalogue.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 """
-This script is run by a Terraform local-exec provisioner to create roles/users in
+This script is run by manually after to create roles/users in
```
Comment (alexwlchan) [inline terraform/shared/scripts/create_elastic_users_catalogue.py:3]:
```suggestion
This script is run manually after "terraform apply" to create roles/users in
```

File: terraform/shared/scripts/create_elastic_users_identity.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+"""
+This script is run by manually after to create roles/users in
```
Comment (alexwlchan) [inline terraform/shared/scripts/create_elastic_users_identity.py:3]:
```suggestion
This script is run manually after "terraform apply" to create roles/users in
```

It'd be good to explain why this is different to the previous script.

File: terraform/shared/scripts/create_elastic_users_identity.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+if __name__ == '__main__':
+    catalogue_role_arn = "arn:aws:iam::756629837203:role/catalogue-developer"
+    catalogue_secret_prefix = f"elasticsearch/catalogue_api"
+
+    username = read_secret(f"{catalogue_secret_prefix}/username", catalogue_role_arn)
+    password = read_secret(f"{catalogue_secret_prefix}/password", catalogue_role_arn)
+
+    role_arn = "arn:aws:iam::770700576653:role/identity-developer"
+    secret_prefix = f"elasticsearch/catalogue_api"
```
Comment (alexwlchan) [inline terraform/shared/scripts/create_elastic_users_identity.py:74]:
```suggestion
    identity_role_arn = "arn:aws:iam::770700576653:role/identity-developer"
    identity_secret_prefix = f"elasticsearch/catalogue_api"
```

to distinguish from the catalogue variables.

File: terraform/shared/scripts/create_elastic_users_identity.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+    es = Elasticsearch(endpoint, http_auth=(username, password))
+
+    # Roles are created already in create_elasticsearch_users_catalogue.py
+
+    # Create usernames
+    newly_created_usernames = []
+    for service_name, roles in SERVICES.items():
+        service_password = secrets.token_hex()
+        # TODO: REMOVE
```
Comment (alexwlchan) [inline terraform/shared/scripts/create_elastic_users_identity.py:90]: Remove what?

**Review by alexwlchan — APPROVED** [review]
Useful README. 👍

File: terraform/shared/scripts/create_elastic_users_identity.py (hunk 5 above)
Comment (kenoir) [reply terraform/shared/scripts/create_elastic_users_identity.py:90]: What was underneath before I removed it. Should have added a TODO reminding me to remove the TODO.

File: terraform/shared/scripts/create_elastic_users_identity.py (hunk 4 above)
Comment (kenoir) [reply terraform/shared/scripts/create_elastic_users_identity.py]: Can't commit this as used later on, will refactor locally.

**Pull request merged**